include/zephyr/rtio/rtio.h

```diff
@@ -269,6 +317,18 @@ struct rtio_executor {
 	const struct rtio_executor_api *api;
 };
 
+/**
+ * @brief An entry mapping a mempool entry to its sqe.
+ */
+struct rtio_mempool_map_entry {
```

Comment:
Could this perhaps be made into a 4 byte struct with a little bit mapping?

```c 
struct rtio_mempool_map_entry {
    bool zombie: 1;
    uint16_t block_idx: 15;
    bool _resv: 1;
    uint16_t block_count: 15;
}
```

As it stands now there's 3 words taken (12 or 24 bytes) in ram for each sqe, doing something like the above would cut it to a 4 byte management entry for each sqe, those add up pretty fast.

At 128 byte blocks with support for 32768 of them, that's a 4 megabyte mem pool.

Reply:
Needed to use the reserved bit to tell if there is an allocation for the sqe, but other than that, this works (I used an enum to keep it simple).